keyboards/keychron/c3_pro/c3_pro.c

```diff
@@ -16,6 +16,47 @@
 
 #include "c3_pro.h"
 
+void keyboard_post_init_kb(void) {
+    setPinOutputPushPull(LED_MAC_OS_PIN);
+    setPinOutputPushPull(LED_WIN_OS_PIN);
+    writePin(LED_MAC_OS_PIN, !LED_OS_PIN_ON_STATE);
+    writePin(LED_WIN_OS_PIN, !LED_OS_PIN_ON_STATE);
+
+    layer_state_t last_layer = eeconfig_read_default_layer();
+    if (last_layer) {
+        default_layer_set(last_layer);
+    } else {
+        default_layer_set(1U << 2);
+    }
```

Comment:
To be honest, I'm not sure what the point of this is. 

default layer is read already:
https://github.com/qmk/qmk_firmware/blob/6a947fa5345b47f8d41c5ab13c0c6dac79cb3bc9/quantum/keyboard.c#L440-L441

This happens in `quantum_init(void)` which happens towards the start of the `keyboard_init(void)` function, and well before dip switch init, and keyboard_post_init_*.

And this appears to call the last default layer, if it's set at all, or to set the default layer to 2 (the 3rd layer), since the default layer state will always have a value. (eeconfig_init sets it to `1<<0` by default, so will always be 1 (0b00000001), at the very least. 

So this really feels like unnecessary code.

Reply:
Looking at the linked code, the layer stuff I drafted from the old PR does seem redundant. I'll try and see what I can switch this to sometime tomorrow so that the code doesn't...suck.

Reply:
Made the change from https://github.com/qmk/qmk_firmware/pull/25049/commits/f3fb6cee3596fd9e898f425e153d026284ee69d9. Will update PR desc. to reflect this.